erpnext/accounts/doctype/pos_invoice/pos_invoice.py

```diff
@@ -212,6 +213,25 @@ def validate_serialised_or_batched_item(self):
 		if error_msg:
 			frappe.throw(error_msg, title=_("Invalid Item"), as_list=True)
 
+	def validate_serial_nos(self):
+		error_msg = []
+		for item in self.get("items"):
+			serialized = item.get("has_serial_no")
+			msg = ""
+			if serialized:
+				valid_serial_nos = frappe.get_all('Serial No', filters={'item_code':item.get("item_code")}, pluck="name")
+				invalid_serials = ""
+				for serial_no in item.get('serial_no').split("\n"):
```

Comment:
Import this function instead of manually splitting it. 

https://github.com/frappe/erpnext/blob/d7b0a47d7cbfcc03322ade1a31d3520caddb6483/erpnext/stock/doctype/serial_no/serial_no.py#L472-L477